app/home/homeController.js

```diff
@@ -8,7 +8,9 @@
 
     function controllerFunction (cityService) {
           var vm = this;
-          vm.city = cityService.city;
+          vm.cityService = cityService;
+          // Curious as to why you need the whole object. Commented out in forecastController.
```

Comment:
I see, so in order to persist the changes you need to pass the entire object or is there a better way?